plugins/meta/aos-firewall/firewall/firewall.go

```diff
@@ -512,3 +671,32 @@ func (f *Firewall) execute(r *iptablesRequest) (err error) {
 
 	return nil
 }
+
+func getInterfaceInfoByIP(ip net.IP) (name string, mask string, err error) {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return "", "", err
+	}
+
+	for _, iface := range interfaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			continue
+		}
+
+		for _, addr := range addrs {
+			iip, ipNet, err := net.ParseCIDR(addr.String())
+			if err != nil {
+				continue
+			}
+
+			if iip.Equal(ip) {
+				prefixLen, _ := ipNet.Mask.Size()
+
+				return iface.Name, strconv.Itoa(prefixLen), nil
+			}
+		}
+	}
+
+	return "", "", fmt.Errorf("couldn't find a interface for the ip: %s", ip.String())
```

Comment:
`a -> an`. But it's ok to omit articles in error messages at all.

Reply:
change to corrct articles